### PR TITLE
Add Brief Command

### PR DESCRIPTION
**Pull request by ransbachm**
#Add Brief Command
Implemented like in the [old bot](https://github.com/drph4nt0m/avbot/blob/73ef652812c3979bfa947a6ccf7b2d4a0dfde493/bot.js#L2154
) and [this similar command](https://github.com/drph4nt0m/avbot-v3/blob/master/src/app/commands/weather/metar.js) for issue #11.

# Questions
There are some I thing I am not sure about read below. The issues should also be commented in the code (if not already solved).
- [x] Is the name ```const *ivao* = require...``` right?
- [x] In what command group does this belong in?
- [x] Should colors (for example for error, or success in embedds) be centralised (like in old bot) or hard coded?
- [x] Why does ```chart = await ivao.download(icao)``` alway return undefined? What am i doing wrong.
- [x] How should the error message look like (or should it just say unavailable like it is now)?

### CODE REVIEW

**Comment by drph4nt0m** [comment]
@ransbachm 

1. Don't use moment-timezone package. For this project I am using dayjs. So change every implementation with dayjs
2. Use charts.ivao.aero for charts. Sample valid chart link: http://charts.avbot.in/VOMM.pdf
3. For getting chart link, create a new class and import it in brief.js 
4. For the footer of the message embed, concat both, see ivao command for example
5. Instead of calling catch on metar class call, surround it with try catch block and move the add field for metar inside the try of it.
6. Rename the class imports to be Capitalized (eg: Avwx, Ivao)
7. You don't need ivao class here.
8. Let the command group be weather itself.
9. Leave the colors as they are for now.

Let me know if you still have any doubts.

**Comment by ransbachm** [comment]
- [ ] Don't use moment-timezone package. For this project I am using dayjs. So change every implementation with dayjs
- [ ] Use charts.ivao.aero for charts. Sample valid chart link: http://charts.avbot.in/VOMM.pdf
- [ ] For getting chart link, create a new class and import it in brief.js
- [ ] For the footer of the message embed, concat both, see ivao command for example
- [ ] Instead of calling catch on metar class call, surround it with try catch block and move the add field for metar inside the try of it.
- [ ] Rename the class imports to be Capitalized (eg: Avwx, Ivao)
- [ ] You don't need ivao class here.
- [ ] Let the command group be weather itself.
- [ ] Leave the colors as they are for now.

**Comment by ransbachm** [comment]
Is the error message ok?

File: src/app/commands/weather/brief.js [hunk 1]
```diff
@@ -1,74 +1,56 @@
...
       aliases: [],
-      description: 'WRITE DESC',
-      examples: ['WRITE DISC'],
+      description: 'Gives you a briefing for the chosen airport.',
+      examples: ['brief <icao>'],
       args: [
         {
           key: 'icao',
-          prompt: 'Second parameter missing 😜 ', //TODO: REPLACE
+          prompt: 'What ICAO would you like the bot to give METAR for?',
```
Comment (drph4nt0m) [inline src/app/commands/weather/brief.js:21]: ... to give briefing for?

File: src/app/commands/weather/brief.js [hunk 2]
```diff
@@ -1,74 +1,56 @@
...
-      .setColor('#0099ff') 
+      .setColor('#0099ff')
+      .setFooter(`${this.client.user.username} • Source: IVAO API`)
+      .setTimestamp();
 
-      // discussion :
-      /* use :?
-      .setFooter('This is not a source for official briefing. Please use the appropriate forums.');
-      then remove comment */
+    const timeform3 = Dayjs.utc().format('YYYY-MM-DD HH:mm:ss [Z]');
```
Comment (drph4nt0m) [inline src/app/commands/weather/brief.js:35]: rename the varibale to zuluTime

File: src/app/commands/weather/brief.js [hunk 3]
```diff
@@ -1,74 +1,56 @@
...
-    console.log("REQ : " + icao);
-
-    const briefEmbed = new Discord.MessageEmbed() //TODO STYLE
+    const briefEmbed = new Discord.MessageEmbed()
       .setTitle(`BRIEF for ${icao.toUpperCase()}`)
-      //TODO: open discussion of color should be centralised
-      // like in old repo (see same file, command), then remove comment
-      .setColor('#0099ff') 
+      .setColor('#0099ff')
+      .setFooter(`${this.client.user.username} • Source: IVAO API`)
```
Comment (drph4nt0m) [inline src/app/commands/weather/brief.js:32]: remove the source,, add "This is not a source for official briefing. Please use the appropriate forums." after the username .

File: src/app/utils/Charts.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const axios = require('axios').default;
+
+module.exports = class Charts {
```
Comment (drph4nt0m) [inline src/app/utils/Charts.js:3]: Rename the class to Chart, update the imports in other places as well

File: src/app/utils/Charts.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    return `https://charts.avbot.in/${name}.pdf`;
+  }
+
+  // returns null if not available, else the correct url
+  static getChartAvailable(name) {
+    return new Promise(async (resolve, reject) => {
+      const response = await this.api.get(`${name}.pdf`).catch((e) => {
+        resolve(null);
+      });
+      return resolve(this.getChartURL(name));
```
Comment (drph4nt0m) [inline src/app/utils/Charts.js:19]: add try catch, and if error/catch it should call reject

**Review by drph4nt0m — CHANGES_REQUESTED** [review]

**Review by drph4nt0m — APPROVED** [review]

**Comment by drph4nt0m** [comment]
@ransbachm thanks for your contribution. Feel free to take up other issues.

**Comment by drph4nt0m** [comment]
@all-contributors please add @ransbachm for code

**Pull request merged**